### DISCUSSION TITLE
Accessing application via coder reverse proxy

### DISCUSSION THREAD

**Discussion by garytaylor** [coder/coder#7616] [General] [answered]
Hi
  I am trying to work out where my problem is - it might be within coder or it might be my setup.

Coder itself is working as normal - I can use the VScode IDE in the web browser, I can use the terminal etc..

However, I have setup an application in the template with the following config (which I copied from an example template for 'Matlab' - as I really didn't know what I was doing)

```
resource "coder_app" "application" {
  agent_id     = coder_agent.main.id
  display_name = "Scout Application"
  slug         = "scout"
  icon         = "https://img.icons8.com/nolan/344/matlab.png"
  url          = "http://${lower(data.coder_workspace.me.name)}.devserver1"
  subdomain    = false
  share        = "public"
}


```

When I click on the link to the application in the workspace, the following error is shown
```
502 - Bad Gateway
Failed to proxy request to application: read tcp [fd7a:115c:a1e0:4a0c:8263:16ab:4023:1cc2]:41597: connection reset by peer
```

I don't understand why it is using that IPV6 address as I am not (intentionally) using IPV6.

I will now try and describe my setup

I have a linux machine running clear linux called 'devserver1'. This is where coder runs (inside docker) and all of my workspaces will run inside this docker instance.
This linux box has its dns to resolve via a 'pihole' (dnsmasq) server to allow *.devserver1 to resolve to 192.168.0.100 (the IP address of devserver1).  Other machines on the network also use this dnsmasq server.  This simply avoids using my public IP address and the traffic having to go via my router etc...

For my application router, I use 'traefik'.  All of my workspaces have the relevant traefik labels added to the docker container to allow connection from inside my network on (workspace-name).devserver1 and is http only, not https (yet)

 Coder itself is also routed via traefik - this is the config from the docker-compose file that runs traefik 

```
    labels:
      - traefik.enable=true
      - traefik.http.routers.coder-coder.rule=HostRegexp(`develop.gdt-software-solutions.co.uk`,`{subhost:[a-zA-Z0-9-]+}.develop.gdt-software-solutions.co.uk`) 
      - traefik.http.routers.coder-coder.entrypoints=https
      - traefik.http.routers.coder-coder.tls.certresolver=myresolver
      - traefik.docker.network=traefik_default 

```

this means that develop.gdt-software-solutions.co.uk or any subdomain will route to coder.  I am not using subdomains yet though.

If you could give me some tips as to how I might find out exactly what is going wrong here, I would really appreciate it

Thanks

**Comment by kylecarbs** [comment]
What version of Coder are you using? We disallow public apps that aren't on a subdomain, but if it is this we're certainly giving you a bad error.

**Comment by matifali** [comment]
when you are not using a subdomain some `coder_app`s require setting very specific `url` paths to be accessible. It depends from application to application.
For example see here, how the paths are set for matlab web. 
https://github.com/matifali/coder-templates/blob/784bcdb6246df49475e8abd2e841df59d89dc455/matlab/main.tf#L62

This is a very old version and is not guaranteed to work. I ended up setting up a wildcard subdomain.

**Comment by matifali** [comment, marked as answer]
You need to set the `url` property of `coder_app` to a url of type `http://localhost:PORT`
